xdsl/irdl.py

```diff
@@ -133,8 +133,18 @@ class AllOf(AttrConstraint):
     """The list of constraints that are checked."""
 
     def verify(self, attr: Attribute) -> None:
+        exc_bucket: list[VerifyException] = []
+
         for attr_constr in self.attr_constrs:
-            attr_constr.verify(attr)
+            try:
+                attr_constr.verify(attr)
+            except VerifyException as e:
+                exc_bucket.append(e)
+
+        if len(exc_bucket):
+            exc_msg = "".join([str(e) + "\n" for e in exc_bucket[:-1]])
```

Comment:
```suggestion
            exc_msg = "\n".join([str(e) for e in exc_bucket])
```
Would that work instead? (and removing the next line)